package/src/components/MessageInput/ShowThreadMessageInChannelButton.tsx

```diff
@@ -1,60 +1,39 @@
-import React from 'react';
-import { StyleSheet, Text, TouchableOpacity, View } from 'react-native';
+import React, { useCallback } from 'react';
+import { Pressable, StyleSheet, Text, View } from 'react-native';
+
+import { MessageComposerState } from 'stream-chat';
 
 import {
-  MessageInputContextValue,
-  useMessageInputContext,
-} from '../../contexts/messageInputContext/MessageInputContext';
+  ChannelContextValue,
+  useChannelContext,
+} from '../../contexts/channelContext/ChannelContext';
+import { useMessageComposer } from '../../contexts/messageInputContext/hooks/useMessageComposer';
 import { useTheme } from '../../contexts/themeContext/ThemeContext';
 import { ThreadContextValue, useThreadContext } from '../../contexts/threadContext/ThreadContext';
 import {
   TranslationContextValue,
   useTranslationContext,
 } from '../../contexts/translationContext/TranslationContext';
+import { useStateStore } from '../../hooks/useStateStore';
 import { Check } from '../../icons';
 
-const styles = StyleSheet.create({
-  checkBox: {
-    alignItems: 'center',
-    borderRadius: 3,
-    borderWidth: 2,
-    height: 16,
-    justifyContent: 'center',
-    width: 16,
-  },
-  container: {
-    flexDirection: 'row',
-    marginHorizontal: 2,
-    marginTop: 8,
-  },
-  innerContainer: {
-    flexDirection: 'row',
-  },
-  text: {
-    fontSize: 13,
-    marginLeft: 12,
-  },
+const stateSelector = (state: MessageComposerState) => ({
+  showReplyInChannel: state.showReplyInChannel,
 });
 
 export type ShowThreadMessageInChannelButtonWithContextProps = Pick<
-  MessageInputContextValue,
-  'sendThreadMessageInChannel' | 'setSendThreadMessageInChannel'
+  ThreadContextValue,
+  'allowThreadMessagesInChannel'
```

Comment:
I have a proposal to deprecate this prop `allowThreadMessagesInChannel` from the ThreadContextValue, as someone who doesn't need the feature can just set the `ShowThreadMessageInChannelButton` to a component returning null. We have these extra props for no good reason. But we can keep it as well if it makes sense.

Reply:
As discussed offline, since this isn't a change that would be useful to both us or any integrator for now we won't be doing it.